edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/ImagePullSecret.cs

```diff
@@ -9,12 +9,41 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment
 
     public class ImagePullSecret
     {
+        public ImagePullSecret(AuthConfig dockerAuth)
+        {
+            this.dockerAuth = dockerAuth;
+            this.Name = ImagePullSecretName.Create(dockerAuth);
+        }
+
+        public ImagePullSecretName Name { get; }
+
+        readonly AuthConfig dockerAuth;
+
+        public string GenerateSecret()
+        {
+            // JSON struct is
+            // { "auths":
```

Comment:
Code cleanup...

Reply:
Nope. This is a useful comment that was left by original author. It describes how JSON structure for ImagePullSecret should look like

Reply:
Got it.